src/magic_folder/web.py

```diff
@@ -136,100 +140,55 @@ class APIv1(Resource, object):
-    _tahoe_client = attr.ib()
-
-    def __attrs_post_init__(self):
-        Resource.__init__(self)
-
-    def getChild(self, name, request):
-        name_u = name.decode("utf-8")
-        try:
-            folder_config = self._global_config.get_magic_folder(name_u)
-        except ValueError:
-            return NoResource(b"{}")
-        return MagicFolderParticipantAPIv1(folder_config, self._tahoe_client)
-
-
-class _InputError(ValueError):
-    """
-    Local errors with our input validation to report back to HTTP
-    clients.
-    """
-
-
-@attr.s
-class MagicFolderParticipantAPIv1(Resource, object):
-    """
-    Implements the v1 API for the ``/participants/<magic-folder-name>``
-    part of the API hierarchy.
-    """
-    _folder_config = attr.ib()
-    _tahoe_client = attr.ib()
-
-    # XXX maybe we could/should pass around a TahoeClient with the
-    # global-config? It could maybe simplify stuff, *so long as* we
-    # can still override a "testing" client into e.g. a "testing"
-    # config...
-
-    def __attrs_post_init__(self):
-        Resource.__init__(self)
-
-    def render_GET(self, request):
+    @app.route("/participants/<string:folder_name>", methods=['GET'])
+    @inlineCallbacks
+    def list_participants(self, request, folder_name):
         """
         List all participants of this folder
         """
+        try:
+            folder_config = self._global_config.get_magic_folder(folder_name)
+        except ValueError:
+            returnValue(NoResource(b"{}"))
+
         collective = participants_from_collective(
-            self._folder_config.collective_dircap,
-            self._folder_config.upload_dircap,
+            folder_config.collective_dircap,
+            folder_config.upload_dircap,
             self._tahoe_client,
         )
-        d = collective.list()
-
-        def listed(participants):
-            reply = {
-                part.name: {
-                    "personal_dmd": part.dircap.decode("ascii"),
-                    # not tracked properly yet
-                    # "public_key": part.verify_key.encode(Base32Encoder),
-                }
-                for part in participants
-            }
-            request.setResponseCode(http.OK)
-            _application_json(request)
-            request.write(json.dumps(reply).encode("utf8"))
-        d.addCallback(listed)
-
-        def failed(reason):
+        try:
+            participants = yield collective.list()
+        except Exception:
             # probably should log this failure
             request.setResponseCode(http.INTERNAL_SERVER_ERROR)
             _application_json(request)
-            request.write(json.dumps({"reason": "unexpected error processing request"}))
-            return None
-        d.addErrback(failed)
-        d.addBoth(lambda ignored: request.finish())
-        return NOT_DONE_YET
+            returnValue(json.dumps({"reason": "unexpected error processing request"}, ensure_ascii=False).encode("utf-8"))
```

Comment:
Why do we need to change `ensure_ascii=` from the default here?

Reply:
I dropped it. I tend to default to setting to false, since everything can deal with utf-8 encoded json, and it makes viewing it by hand easier.